roles/bmc_fw_update/tasks/main.yml

```diff
@@ -1,2 +1,85 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2022-2024 Dell Inc, or its subsidiaries.
+
 ---
 # tasks file for bmc_fw_update
+
+- name: Get Firmware Inventory
+  community.general.redfish_info:
+    category: Update
+    command: GetFirmwareInventory
+    baseuri: "{{ inventory_hostname }}"
+    username: "{{ dpu_bmc_username }}"
+    password: "{{ dpu_bmc_password }}"
+  register: fw_inventory_before
+
+- name: Print BMC Version
+  ansible.builtin.debug:
+    msg: "{{ fw_inventory_before.redfish_facts.firmware.entries[0].Version }}"
+
+- name: Download firmware {{ firmware.bmc }} image
+  ansible.builtin.get_url:
+    url: "{{ firmware.url }}/{{ firmware.bmc }}"
+    dest: /tmp/{{ firmware.bmc }}
+    mode: '0440'
+
+- name: Update BMC firmware of DPU
+  community.general.redfish_command:
+    category: Update
+    command: MultipartHTTPPushUpdate
+    baseuri: "{{ inventory_hostname }}"
+    username: "{{ dpu_bmc_username }}"
+    password: "{{ dpu_bmc_password }}"
+    timeout: 600
+    update_image_file: "/tmp/{{ firmware.bmc }}"
+  register: result_update_task
+
+- name: Print TASK id for tracking
+  ansible.builtin.debug:
+    msg: "{{ result_update_task.return_values.update_status.handle }}"
+
+- name: Pause for 10 seconds for BMC to start task
+  ansible.builtin.pause:
+    seconds: 10
+
+- name: Get the status of an update operation in a loop
+  community.general.redfish_info:
+    category: Update
+    command: GetUpdateStatus
+    baseuri: "{{ inventory_hostname }}"
+    username: "{{ dpu_bmc_username }}"
+    password: "{{ dpu_bmc_password }}"
+    update_handle: "{{ result_update_task.return_values.update_status.handle }}"
+  register: update_progress
+  until: update_progress.redfish_facts.update_status.status != 'Running'
+  retries: 60
+  delay: 30
+
+- ansible.builtin.assert: { that: "update_progress.redfish_facts.update_status.status == 'Completed'" }
+
+- name: Reboot BMC to apply new firmware of DPU
```

Comment:
This can be different reset/reboot bmc role with available options and can be reused here.

Reply:
done. am I using `include_role` correctly ?